livekit-agents/livekit/agents/utils/av_sync.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+import logging
+import time
+from collections import deque
+from typing import Optional, Union
+
+import livekit.agents.utils as utils
+from livekit import rtc
+
+logger = logging.getLogger(__name__)
+
+
+class AVSynchronizer:
+    """Synchronize audio and video capture.
+
+    Usage:
+        av_sync = AVSynchronizer(
+            audio_source=audio_source,
+            video_source=video_source,
+            video_sample_rate=video_sample_rate,
+        )
+
+        async for video_frame, audio_frame in video_generator:
+            av_sync.push(video_frame)
+            av_sync.push(audio_frame)
+    """
+
+    def __init__(
+        self,
+        *,
+        audio_source: Optional[rtc.AudioSource],
+        video_source: Optional[rtc.VideoSource],
+        video_sample_rate: Optional[float],
```

Comment:
nit: better to call it fps

```suggestion
        video_fps: Optional[float],
```